cluster/replication/shard_replication_apply.go

```diff
@@ -62,7 +61,12 @@ func (s *ShardReplicationFSM) RegisterError(c *api.ReplicationRegisterErrorReque
 // is held
 func (s *ShardReplicationFSM) writeOpIntoFSM(op ShardReplicationOp, status ShardReplicationOpStatus) error {
 	if _, ok := s.opsByTargetFQDN[op.TargetShard]; ok {
-		return ErrShardAlreadyReplicating
+		// First check the status of the existing op. If it's READY or CANCELLED we can accept a new op
+		if existingOpStatus, ok := s.statusById[op.ID]; ok {
+			if existingOpStatus.GetCurrentState() != api.READY && existingOpStatus.GetCurrentState() != api.CANCELLED {
```

Comment:
I think an op can be in the cancelled state but still finishing up the "clean up". I don't think we have a state signal that indicates when a op has finished the cancellation process, but I could be wrong. Could we change this to just accepting ops in the READY state? I would be ok with requiring the operator to manually delete any ops which ended up in the cancelled state before they recreate/rerun the plan.